File: packages/backend/src/main/kotlin/quizzem/backend/features/question/api/dto/CreateQuestionDto.kt
package quizzem.backend.features.question.api.dto

import quizzem.backend.features.question.model.AnswerMode
import quizzem.backend.features.question.model.MediaType

data class CreateQuestionDto(
    val text: String,
    val answerMode: AnswerMode? = AnswerMode.SINGLE_CHOICE,
    val mediaType: MediaType? = MediaType.NONE,
    val answers: List<CreateAnswerDto> = emptyList()
)

File: packages/backend/src/features/question/api/dto/create-question.dto.ts
import { Type } from 'class-transformer';
import {
  ArrayNotEmpty,
  IsEnum,
  IsString,
  ValidateNested,
} from 'class-validator';
import { CreateAnswerDto } from 'src/features/question/api/dto/create-answer.dto';
import {
  type AnswerMode,
  AnswerModeEnum,
} from 'src/features/question/model/answer-mode';
import {
  type MediaType,
  MediaTypeEnum,
} from 'src/features/question/model/media-type';

export class CreateQuestionDto {
  @IsString()
  text!: string;

  @IsEnum(AnswerModeEnum)
  answerMode: AnswerMode = AnswerModeEnum.SINGLE_CHOICE;

  @IsEnum(MediaTypeEnum)
  mediaType: MediaType = MediaTypeEnum.NONE;

  @ArrayNotEmpty()
  @ValidateNested({ each: true })
  @Type(() => CreateAnswerDto)
  answers: CreateAnswerDto[] = [];
}
